### PR TITLE
Admin login - check that auth code state hasn't changed (protect from XSRF)

### LINKED ISSUE #512: Admin Login - properly check that auth code state hasn't changed
# Description
In the admin project AuthController, we need to check that the auth code state is what we expected, in order to [protect](https://developers.google.com/identity/protocols/oauth2/openid-connect#createxsrftoken) from a XSRF attack.

To do this, we must upgrade from Lumen to Laravel so that we can save the state string in session.

### PR DESCRIPTION
**Pull request by tristan-orourke**
Resolves #512 

Upgrades Admin project from Lumen to Laravel in order to save some Session data.

Note: #471 MUST BE MERGED BEFORE THIS PR.

### CODE REVIEW

File: admin/app/Models/User.php [hunk 1]
```diff
@@ -2,28 +2,28 @@
 
 namespace App\Models;
 
-use Illuminate\Auth\Authenticatable;
-use Illuminate\Contracts\Auth\Access\Authorizable as AuthorizableContract;
-use Illuminate\Contracts\Auth\Authenticatable as AuthenticatableContract;
+use Illuminate\Contracts\Auth\MustVerifyEmail;
```
Comment (mnigh) [inline admin/app/Models/User.php:5]: Symbol 'App\Models\MustVerifyEmail' is declared but not used.

File: admin/app/Models/User.php [hunk 2]
```diff
@@ -2,28 +2,28 @@
...
-use Laravel\Lumen\Auth\Authorizable;
+use Illuminate\Foundation\Auth\User as Authenticatable;
+use Illuminate\Notifications\Notifiable;
+use Laravel\Sanctum\HasApiTokens;
 
-class User extends Model implements AuthenticatableContract, AuthorizableContract
+class User extends Authenticatable
 {
-    use Authenticatable, Authorizable, HasFactory;
+    use HasApiTokens, HasFactory, Notifiable;
```
Comment (mnigh) [inline admin/app/Models/User.php:13]: Undefined type 'Laravel\Sanctum\HasApiTokens'.

File: admin/app/Providers/AuthServiceProvider.php [hunk 3]
```diff
@@ -2,38 +2,29 @@
 
 namespace App\Providers;
 
-use App\Models\User;
+use Illuminate\Foundation\Support\Providers\AuthServiceProvider as ServiceProvider;
 use Illuminate\Support\Facades\Gate;
```
Comment (mnigh) [inline admin/app/Providers/AuthServiceProvider.php:6]: Symbol 'App\Providers\Gate' is declared but not used.

File: admin/app/Providers/EventServiceProvider.php [hunk 4]
```diff
@@ -2,7 +2,10 @@
 
 namespace App\Providers;
 
-use Laravel\Lumen\Providers\EventServiceProvider as ServiceProvider;
+use Illuminate\Auth\Events\Registered;
+use Illuminate\Auth\Listeners\SendEmailVerificationNotification;
+use Illuminate\Foundation\Support\Providers\EventServiceProvider as ServiceProvider;
+use Illuminate\Support\Facades\Event;
```
Comment (mnigh) [inline admin/app/Providers/EventServiceProvider.php:8]: Symbol 'App\Providers\Event' is declared but not used.

**Review by mnigh — COMMENTED** [review]
Nice work. A few minor issues and then ready to be merged.

File: admin/app/Models/User.php (hunk 1 above)
Comment (tristan-orourke) [reply admin/app/Models/User.php:5]: I'm tempted to just delete User entirely, but worried that might break something in Laravel. I'll just apply your recommendations for now.

**Review by mnigh — APPROVED** [review]
👍

**Pull request merged**